Added filter for no stock notification emails

jondcampbell opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

Added one new filter so we can do a check before sending the no stock notification. We have some subscription products that are out of stock and we want renewals to work properly but to not have email notifications happening every time the orders go through. This is a kinda related to (and based on): https://github.com/woocommerce/woocommerce/pull/26908


### How to test the changes in this Pull Request:

1. Use the filter to stop the no stock notification emails from happening

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [] Have you successfully run tests with your changes locally?

### Changelog entry
> New - Add `woocommerce_should_send_no_stock_notification` filter

jondcampbell committed:
add no stock notification filter

jondcampbell committed:
Update class-wc-emails.php

removing whitespace to be coding standards compliant.

rrennick reviewed (changes requested):
This should be good to go with the requested changes.

jondcampbell committed:
simplifying filter usage

Switched to product id instead of product object
Moving the filter into the if condition instead of having a single use variable

jondcampbell commented on includes/class-wc-emails.php:
```diff
@@ -638,6 +638,18 @@ public function no_stock( $product ) {
+		 *
+		 * @since 4.4.0
+		 */
+		$should_notify = apply_filters( 'woocommerce_should_send_no_stock_notification', true, $product );
```
@rrennick I've made the requested changes.

rrennick reviewed (approved):
@jondcampbell Thanks for the quick turn around on the changes. 

**Note** Before merging this the @since will need to be updated to the appropriate version.

rodrigoprimo reviewed (approved):
Looks good to me. Thanks for your contribution.

rodrigoprimo commented:
I will merge this and update the version in the `@since` tag in a separate commit.

rodrigoprimo commented:
Version updated in commit d99053dbedcaaca04c8ecbb8c8bea6848ee40c19
